cmake/modules/BuildPlugin.cmake

```diff
@@ -73,7 +73,11 @@ MACRO(BUILD_PLUGIN PLUGIN_NAME)
 	INSTALL(TARGETS ${PLUGIN_NAME} LIBRARY DESTINATION "${PLUGIN_DIR}")
 
 	IF(LMMS_BUILD_APPLE)
-		SET_TARGET_PROPERTIES(${PLUGIN_NAME} PROPERTIES LINK_FLAGS "-bundle_loader \"${CMAKE_BINARY_DIR}/lmms\"")
+		IF ("${PLUGIN_LINK}" STREQUAL "DYNAMIC")
```

Comment:
I don't think it will make sense to use `-bundle_loader` for `SHARED` on macOS. How about keep using `SHARED` and testing for it instead of the new `DYNAMIC`?

Reply:
@PhysSong I still think we can do this cleaner.  Should we just have a static list of `.dylib`s that are in `BuildPlugin.cmake` and use that for toggling `-undefined dynamic_lookup`?  I hate the additional non-standard `DYNAMIC` link flag more and more each time I stare at it.

Reply:
Furthermore, if this is the only instance where we call `SHARED`, I'd be fine switching the default behavior.

Reply:
I'm 99% sure this is the only instance of `SHARED` since it was explicitly added for this library in #2578.  Cleaned up via f640b8e.